lib/pbench/server/api/resources/__init__.py

```diff
@@ -1759,28 +1779,35 @@ def _delete(
         )
 
     @Auth.token_auth.login_required(optional=True)
-    def get(self, **kwargs):
+    def get(self, **kwargs) -> Response:
         """
         Handle an authenticated GET operation on the Resource
         """
         return self._dispatch(ApiMethod.GET, kwargs)
 
     @Auth.token_auth.login_required(optional=True)
-    def post(self, **kwargs):
+    def head(self, **kwargs) -> Response:
+        """
+        Handle an authenticated HEAD operation on the Resource
+        """
+        return self._dispatch(ApiMethod.HEAD, kwargs)
```

Comment:
Does this work? This worries me, and seems like we no longer have `HEAD` implementations for our `GET` APIs. Normally, Flask implements a `HEAD` as a `GET` and discards the response payload (returning only headers). I don't think it does that if the `head` method is implemented, which you're now doing anywhere, so I think a `HEAD` ought to now go through to the fallback (not implemented` "abstract" `_head`.

While I'm not against the idea of being *able* to implement a specialized `_head` I don't think this way will work.

Or at least, I know that a `HEAD /api/v1/datasets/list` worked before, and I'm not confident that it'll work with these changes. (In fact, I don't see how it could.)

Reply:
I'll take a look.

Reply:
Yes, you are correct that a `HEAD` operation is not working now.  I knew the client allowed it, but didn't think twice about this because we have no tests that exercise the `HEAD` operation.  I have added one to the functional tests in PR #3123.

I am not a fan of this Flask behavior.  For certain APIs we really want a separate method for the `HEAD` operation so that it does not consume server side resources unnecessarily.  Pulling a large tar ball or a large query set can be costly.

I'll noodle about this and post a PR with a suggested way to address this.